frontend/src/components/Field/Rating/Rating.tsx

```diff
@@ -0,0 +1,186 @@
+import { Fragment, useEffect, useMemo, useState } from 'react'
+import {
+  Box,
+  forwardRef,
+  Grid,
+  Stack,
+  Text,
+  useBreakpointValue,
+  Wrap,
+  WrapItem,
+} from '@chakra-ui/react'
+
+import { FieldColorScheme } from '~theme/foundations/colours'
+
+import { RatingOption } from './RatingOption'
+
+export interface RatingProps {
+  /**
+   * The `name` attribute forwarded to each rating `radio` element
+   */
+  name: string
+  /**
+   * Number of rating options to render.
+   */
+  numberOfRatings: number
+  /**
+   * Color scheme of the component to render. Defaults to `primary`.
+   */
+  colorScheme?: FieldColorScheme
+  /**
+   * Function called once a rating is selected.
+   * @param newRating the value of the checked radio
+   */
+  onChange?: (newRating: number | undefined) => void
+  /**
+   * Number of rating components to show per row when unable to display all
+   * components on a single line. Defaults to `5`.
+   */
+  wrapComponentsPerRow?: number
+  /**
+   * The value of the rating to be `checked` initially.
+   */
+  defaultValue?: number
+
+  /**
+   * Variant of rating field to render
+   */
+  variant: 'Heart' | 'Star' | 'Number'
+
+  /**
+   * Helper text to be displayed to quantify the ratings, if any.
+   */
+  helperText?: string
+}
+
+export const Rating = forwardRef<RatingProps, 'input'>(
+  (
+    {
+      colorScheme = 'primary',
+      defaultValue,
+      name,
+      numberOfRatings,
+      onChange,
+      variant,
+      wrapComponentsPerRow = 5,
+      helperText,
+    },
+    ref,
+  ) => {
+    /**
+     * Process given defaultValue and prevents invalid numbers from being used.
+     * @param val the number to process
+     * @returns given value if value is in range [1, numberOfRatings], undefined otherwise.
+     */
+    const processDefaultValue = (val: number | undefined) => {
+      if (!val || val > numberOfRatings || val < 1) {
+        return undefined
+      }
+      return val
+    }
+
+    const [currentValue, setCurrentValue] = useState<number | undefined>(
+      processDefaultValue(defaultValue),
+    )
+
+    // Call onChange everytime currentValue changes.
+    useEffect(() => {
+      onChange?.(currentValue)
+    }, [currentValue, onChange])
```

Comment:
instead of relying on `setCurrentValue` to set the value and calling `onChange` as a side effect, why not declare a wrapper function which calls both `setCurrentValue` and `onChange`, and pass that as the `onChange` prop to `RatingOption`? doesn't that make the change process more explicit?

Reply:
you right